File: SchoolWebApp-API/Project.Core/Entities/Academics/SubjectGroup.cs
﻿using SchoolWebApp.Core.Entities.Shared;
using System.ComponentModel.DataAnnotations;

namespace SchoolWebApp.Core.Entities.Academics
{
    public class SubjectGroup : Base
    {
        [Required(ErrorMessage = "Enter the subject group")]
        [Display(Name = "Subject group name")]
        [StringLength(255)]
        public required string Name { get; set; }
        public required string Abbreviation { get; set; }
        public string? Description { get; set; }
        public int Rank { get; set; }
        public int CurriculumId { get; set; }
        public Curriculum? Curriculum { get; set; }

        public List<Subject> Subjects { get; set; } = new();
    }
}

File: SchoolWebApp-Client/src/app/academics/models/subject-group.ts
import { ResourceModel } from "@/core/models/ResourceModel";
import { Curriculum } from "./curriculum";

export class SubjectGroup extends ResourceModel<SubjectGroup> {
    public name?: string;
    public description?: string;
    public abbreviation?: string;

    public curriculumId?: number;
    public curriculum?: Curriculum;

    constructor(model?: Partial<SubjectGroup>) {
        super(model);
      }
}